src/Shared/AppModels/AppDebug.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Input;
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+using Microsoft.Extensions.Logging;
+using OwlCore.Diagnostics;
+using Windows.ApplicationModel.Core;
+using Windows.Storage;
+using Windows.System;
+using Windows.System.Diagnostics;
+using Windows.UI.Popups;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Core;
+
+namespace StrixMusic.AppModels
+{
+    /// <summary>
+    /// Responsible for handling debug information.
+    /// </summary>
+    public partial class AppDebug : ObservableObject
+    {
+        private SemaphoreSlim _semaphoreSlim = new SemaphoreSlim(1, 1);
+
+        [ObservableProperty]
+        private string? _windowSizeStr;
+
+        [ObservableProperty]
+        private string? _memoryUsage;
+
+        /// <summary>
+        /// Holds the list of all logs.
+        /// </summary>
+        public ObservableCollection<string>? AppLogs { get; } = new ObservableCollection<string>();
+
+        /// <summary>
+        /// Command to delete user settings and data.
+        /// </summary>
+        public ICommand PerformUserDataDeletionCommand { get; private set; }
+
+        /// <summary>
+        /// Creates a new instance of <see cref="AppDebug"/>.
+        /// </summary>
+        public AppDebug()
+        {
+            Logger.MessageReceived += Logger_MessageReceived;
+            Window.Current.SizeChanged += Current_SizeChanged;
+            MemoryManager.AppMemoryUsageIncreased += MemoryManager_MemoryChanged;
+            MemoryManager.AppMemoryUsageDecreased += MemoryManager_MemoryChanged;
+            PerformUserDataDeletionCommand = new RelayCommand(PerformUserDataDeletion);
+            UpdateMemoryUsage();
+            SetWindowSize();
+        }
+
+        private void MemoryManager_MemoryChanged(object sender, object e)
+        {
+            UpdateMemoryUsage();
+        }
+
+        private void UpdateMemoryUsage()
+        {
+            var memoryUsed = MemoryManager.AppMemoryUsage;
+            var totalMemory = MemoryManager.AppMemoryUsageLimit;
+            MemoryUsage = $"App Memory used: {SizeSuffix((long)memoryUsed)}, Expected app memory usage limit: {SizeSuffix((long)totalMemory)}.";
```

Comment:
Let's create separate `MemoryUsed` and `TotalMemory` properties instead of using a string